next-tavla/app/(admin)/edit/[id]/components/TileCard/TransportModeAndLines.tsx

```diff
@@ -0,0 +1,103 @@
+'use client'
+import { TTransportMode } from 'types/graphql-schema'
+import { TTile } from 'types/tile'
+import { TransportIcon } from 'components/TransportIcon'
+import { transportModeNames } from './utils'
+import { Checkbox } from '@entur/form'
+import { TLineFragment } from './types'
+import { useState } from 'react'
+
+function TransportModeAndLines({
+    tile,
+    transportMode,
+    lines,
+}: {
+    tile: TTile
+    transportMode: TTransportMode | null
+    lines: TLineFragment[]
+}) {
+    const lineElements = document.getElementsByName(
+        `${tile.uuid}-${transportMode}`,
+    )
+
+    const anyLineInWhitelist = lines.some((l) =>
+        tile.whitelistedLines?.includes(l.id),
+    )
+    const missingLinesInWhitelist = lines.some(
+        (l) => !tile.whitelistedLines?.includes(l.id),
+    )
+
+    const defaultChecked = () => {
+        if (missingLinesInWhitelist && anyLineInWhitelist)
+            return 'indeterminate'
+        return (
+            !tile.whitelistedLines ||
+            tile.whitelistedLines.length === 0 ||
+            !missingLinesInWhitelist
+        )
+    }
+    const [checked, setChecked] = useState<boolean | 'indeterminate'>(
+        defaultChecked(),
+    )
+
+    return (
+        <div>
+            <div className="flex flex-row gap-4 items-center justify-start font-semibold">
+                <TransportIcon
+                    transportMode={transportMode}
+                    className="w-8 h-8"
+                />
+                {transportModeNames(transportMode)}
+            </div>
+            <div className="border-b-secondary border-b-2 my-2" />
+            <div className="flex flex-row items-center">
+                <Checkbox
+                    checked={checked}
+                    onChange={(e) => {
+                        setChecked(e.currentTarget.checked)
+                        lineElements.forEach((input) => {
+                            if (input instanceof HTMLInputElement)
+                                input.checked = e.currentTarget.checked
+                        })
+                    }}
+                >
+                    Velg alle
+                </Checkbox>
+            </div>
+            {lines.map((line) => (
+                <Checkbox
+                    name={`${tile.uuid}-${transportMode}`}
+                    defaultChecked={
+                        !tile.whitelistedLines ||
+                        tile.whitelistedLines.length === 0 ||
```

Comment:
Er det meningen at når man har trykket av for at man ikke skal vise _noen_ linjer, så blir de automatisk satt til at de er sjekket av?

Reply:
ref kommentar over også

Reply:
Ja, det er siden linjer er en whitelist. Hvis ingen linjer er valgt vil alle linjer vises. For å vise brukeren korrekt informasjon må vi gjøre det sånn